### PR TITLE
Chs

### PR DESCRIPTION
**Pull request by JavaLavaMT**
WIP

### CODE REVIEW

File: lib/models/hub-registration/middleware/trainee-creation.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+          if (!hubParticipant.ssn) {
+            return;
+          }
+
+          let trainee = (await Trainee.find({
+            ssn: hubParticipant.ssn
+          }))[0];
+
+          hubParticipantIds.push(hubParticipant._id);
```
Comment (JavaLavaMT) [inline lib/models/hub-registration/middleware/trainee-creation.js:24]: hubParticipant._id or hubParticipant.id??

File: lib/serializers/hub-registration/index.js [hunk 2]
```diff
@@ -15,6 +15,7 @@ exports.serialize = function (record) {
   if (record.creator){
     record.creator = record.creator._id;
   }
+
```
Comment (JavaLavaMT) [inline lib/serializers/hub-registration/index.js:18]:
```suggestion
```

File: lib/models/hub-registration/adapter.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 exports.create = function*(request, transaction, hash) {
   // Insert registration header
   let reg = yield this.record.constructor.populate(this.record, [ {
-    path: 'participants'
+    path: 'hubParticipants'
```
Comment (JavaLavaMT) [inline lib/models/hub-registration/adapter.js:4]: I think we still need participants

File: lib/serializers/hub-registration/index.js (hunk 2 above)
Comment (Evilscaught) [reply lib/serializers/hub-registration/index.js:18]: ✅

File: lib/models/hub-registration/adapter.js (hunk 3 above)
Comment (Evilscaught) [reply lib/models/hub-registration/adapter.js:4]:
```suggestion 
  {
    path: 'participants'
  }, {
    path: 'hubParticipants'
  },
```

File: lib/models/hub-registration/middleware/trainee-creation.js (hunk 1 above)
Comment (Evilscaught) [reply lib/models/hub-registration/middleware/trainee-creation.js:24]: _id ✅ 
Screenshot 2024-01-25 at 2 54 07 PM

File: lib/models/hub-registration/adapter.js (hunk 3 above)
Comment (Evilscaught) [reply lib/models/hub-registration/adapter.js]: ✅

File: lib/models/hub-registration/middleware/trainee-creation.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+const HubClass = require('../../hub-class');
+const HubParticipant = require('../../hub-participant/index');
+
+async function middleware (next) {
+  if (this.isNew) {
+    const hubClass = await HubClass.findById(this.hubClass);
+    const hubParticipantIds = []
+    
+    if (hubClass.f2f) {
+      if (Array.isArray(this.hubParticipants)) {
```
Comment (JavaLavaMT) [inline lib/models/hub-registration/middleware/trainee-creation.js:11]: can you combine these two

Comment (Evilscaught) [reply lib/models/hub-registration/middleware/trainee-creation.js]: Combined! ✅

**Pull request merged**